Cython/Debugger/libcython.py

```diff
@@ -117,6 +117,62 @@ def wrapper(self, string, *args, **kwargs):
 # Classes that represent the debug information
 # Don't rename the parameters of these classes, they come directly from the XML
 
+def simple_repr(self, renamed=None, skip=(), state=True):
+    """Prints out all instance variables needed to recreate an object.
+
+    Following the python convention for __repr__, this function prints all the
+    information stored in an instance as opposed to its class. The working
+    assumption is that most initialization arguments are stored as a property
+    using the same name.
+
+    The object contents are displayed as the initialization call followed by,
+    optionally, the value of each of the instance's properties in the form:
+    ```
+    ClassName(
+            init_arg_1 = "repr of some example str",
+            ...
+        )
+    self.state_based_property = ...
+    ```
+
+    Function arguments:
+    self        Instance to be represented
+
+    renamed     Dictionary of initialization arguments that are stored under a
+                different property name in the form { argument: property }
+
+    skip        List of instance variables not to be listed as properties.
+                Computed properties are already skipped.
+                Only changes behavior if `state` argument is True (default).
+
+    state       Boolean representing whether properties outside the
+                initialization parameters should be printed (self.prop = ...).
+                Using `False` may make the class more amenable to recursive repr
+    """
+    import inspect
+    init_arg_names = tuple(inspect.signature(self.__init__).parameters)
+    init_attrs = tuple(renamed.get(arg, arg) for arg in init_arg_names) \
+            if renamed else init_arg_names
```

Comment:
There isn't really a reason for this to be a tuple, is there?
```suggestion
    init_attrs = [renamed.get(arg, arg) for arg in init_arg_names] if renamed else init_arg_names
```